### DISCUSSION TITLE
generating table dynamically

### DISCUSSION THREAD

**Discussion by spyshow** [directus/directus#12617] [Archive]
hi all

I have an app that logs some data for several machines which I can add through an admin panel, each machine should have a separate table 
just wondering if I can do this with directus : 
if the user press adds machine in the admin panel it adds it to the machines table and then adds a new table with the name of this machine.

**Comment by rijkvanzanten** [comment]
You could definitely do that in a custom hook! On items.create for your `machines` collection, you could use the CollectionsService (docs coming very soon, but brief example below) to automatically generate a new table:

```js
const { CollectionsService } = services;

await collectionsService.createOne({
  collection: 'my-new-collection',
  schema: {},
  meta: {}
});
```

**Reply by spyshow** [reply]
Great . Can directus generate an api for it automatically??

**Reply by rijkvanzanten** [reply]
Every collection in the system gets an API endpoint, so that's all handled automagically 🙂

**Reply by alexdabast** [reply]
> You could definitely do that in a custom hook! On items.create for your `machines` collection, you could use the CollectionsService (docs coming very soon, but brief example below) to automatically generate a new table:
> 
> ```js
> const { CollectionsService } = services;
> 
> await collectionsService.createOne({
>   collection: 'my-new-collection',
>   schema: {},
>   meta: {}
> });
> ```

any news on the documentation ?

**Reply by azrikahar** [reply]
> any news on the documentation ?

@alexdabast It is currently being tracked in directus/docs#20 👍
